core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -766,11 +766,27 @@ class WorkspaceService(protected val userInfo: UserInfo,
             }
           }
           workspaces.map { workspace =>
+
+            val maybeMcWorkspace : Option[WorkspaceDescription] =
+             workspace.workspaceType match {
+              case WorkspaceType.McWorkspace =>
+                Option(workspaceManagerDAO.getWorkspace(workspace.workspaceIdAsUUID, userInfo.accessToken))
+              case WorkspaceType.RawlsWorkspace => None
+            }
+
+            val cloudPlatform = maybeMcWorkspace match {
```

Comment:
You can add conditional clauses to the case statements. This can give you a more clear control flow, and helps the compiler check that you've logically covered all of your possibilities. Once you start using the if statements, those guarantees go away.  

```suggestion
            val cloudPlatform0 = workspace.workspaceType match {
              case WorkspaceType.McWorkspace => Option(workspaceManagerDAO.getWorkspace(workspace.workspaceIdAsUUID, userInfo.accessToken)) match {
                case Some(mcWorkspace) if (mcWorkspace.getAzureContext != null) =>  Option(WorkspaceCloudPlatform.Azure)
                case Some(mcWorkspace) if (mcWorkspace.getGcpContext != null) => Option(WorkspaceCloudPlatform.Gcp)
                case _ =>  throw new RawlsException(s"unexpected state, no cloud context found for workspace ${workspace.workspaceId}")
              }
              case WorkspaceType.RawlsWorkspace => Option(WorkspaceCloudPlatform.Gcp)
            }
```